### PR TITLE
nobody resource paths based on policy definitions

### PR DESCRIPTION
**Pull request by przemyslaw-dobrowolski-cl**
A new implementation for resource path whitelists. This one is based on a policy configuration defined in schema file. When a whitelisted resource is accessed it is seen as being read by nobody user but does not require authorization.

### CODE REVIEW

File: server/middleware/middleware.go [hunk 1]
```diff
@@ -170,16 +222,29 @@ func Authentication() martini.Handler {
...
 		}
+
 		authToken := req.Header.Get("X-Auth-Token")
-		if authToken == "" {
-			HTTPJSONError(res, "No X-Auth-Token", http.StatusUnauthorized)
-			return
+
+		var targetIdentityService IdentityService
+
+		if len(authToken) == 0 {
```
Comment (nati) [inline server/middleware/middleware.go]: nit:  autoToken == ""

Comment (przemyslaw-dobrowolski-cl) [inline server/middleware/middleware.go]: fixed

**Comment by nati** [comment]
please rebase commit

**Comment by przemyslaw-dobrowolski-cl** [comment]
rebased & squashed

**Comment by nati** [comment]
Thanks! good job

**Pull request merged**